### PR TITLE
Document i2gw interpretation of Ingress annotations/fields

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Documents some of the *current* interpretation of Ingress fields and annotations, which may change over time

Also documents some "ideals" for conflict resolution, that are open for discussion!

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
Just WIP so far, and added to the README, can move it to somewhere else we think is better if needed!

File: README.md [hunk 1]
```diff
@@ -44,7 +73,7 @@ support for the following annotations:
 * nginx.ingress.kubernetes.io/canary-weight-total
 
 If you are reliant on any annotations not listed above, you'll need to manually
-find a Gateway API equivalent. 
+find a Gateway API equivalent.
```
Comment (sunjayBhatia) [inline README.md:76]: still WIP, need to document how annotations should be treated

File: README.md [hunk 2]
```diff
@@ -27,6 +27,35 @@ the following within the repo:
...
+## Conversion of Ingress resources to Gateway API
+
+### Processing Order and Conflicts
+
+Ingress resources will be processed with a defined order to ensure deterministic generated Gateway API configuration.
+This should also determine precedence order of Ingress resources and routes in case of conflicts.
+
+Ingress resources with the oldest creation timestamp will be sorted first and therefore given precedence.
+If creation timestamps are equal, then sorting will be done based on the namespace/name of the resources.
+If an Ingress rule conflicts with another (e.g. same path match but different backends) an error will be reported for the one that sorted later.
```
Comment (youngnick) [inline README.md:39]: While really like this process, I feel like I should mention that some users expect the ingress-nginx behavior of merging matching Ingress objects into load-balanced sets. I'm okay with not doing this for now, but we need to remember it for later.

File: README.md (hunk 1 above)
Comment (youngnick) [reply README.md:76]: I agree that we shouldn't translate annotations at all - we want to discourage annotation use on Gateway API objects as much as possible.

**Review by youngnick — APPROVED** [review]
LGTM

**Comment by youngnick** [comment]
I couldn't remember if we were Prow-ing in this repo. I'll approve and let the next person lgtm

/approve

File: README.md [hunk 3]
```diff
@@ -27,24 +27,53 @@ the following within the repo:
...
-* nginx.ingress.kubernetes.io/canary-by-header
-* nginx.ingress.kubernetes.io/canary-by-header-value
-* nginx.ingress.kubernetes.io/canary-by-header-pattern
-* nginx.ingress.kubernetes.io/canary-weight
+* nginx.ingress.kubernetes.io/canary: If set to `true` will enable weighting backends.
+* nginx.ingress.kubernetes.io/canary-by-header: If specified, the value of this annotation is the header name that will be added as a HTTPHeaderMatch for the routes generated from this Ingress. If not specified, no HTTPHeaderMatch will be generated.
+* nginx.ingress.kubernetes.io/canary-by-header-value: If specified, the value of this annotation is the header value to perform an `HeaderMatchExact` match on in the generated HTTPHeaderMatch.
+* nginx.ingress.kubernetes.io/canary-by-header-pattern: If specified, this is the  pattern to match against for the HTTPHeaderMatch, which will be of type `HeaderMatchRegularExpression`.
+* nginx.ingress.kubernetes.io/canary-weight: If specified and non-zero, this value will be applied as the weight of the backends for the routes generated from this Ingress resource.
 * nginx.ingress.kubernetes.io/canary-weight-total
```
Comment (sunjayBhatia) [inline README.md:73]: didn't add anything for this one as I don't see it being used in practice in the code?

Comment (robscott) [reply README.md:73]: Oh good catch, I'll create an issue to add that.

**Review by robscott — APPROVED** [review]
Thanks @sunjayBhatia!

**Pull request merged**